app/Http/Controllers/HafasController.php

```diff
@@ -171,25 +172,67 @@ private static function fetchHafasTrip(string $tripID, string $lineName): HafasT
 
         $polylineHash = TransportController::getPolylineHash(json_encode($tripJson->polyline))->hash;
 
-        return HafasTrip::updateOrCreate([
-                                             'trip_id' => $tripID
-                                         ], [
-                                             'category'    => $tripJson->line->product,
-                                             'number'      => $tripJson->line->id,
-                                             'linename'    => $tripJson->line->name,
-                                             'origin'      => $origin->ibnr,
-                                             'destination' => $destination->ibnr,
-                                             'stopovers'   => json_encode($tripJson->stopovers),
-                                             'polyline'    => $polylineHash,
-                                             'departure'   => $tripJson->plannedDeparture,
-                                             'arrival'     => $tripJson->plannedArrival,
-                                             'delay'       => $tripJson->arrivalDelay ?? null
-                                         ]);
+        $hafasTrip = HafasTrip::updateOrCreate([
+                                                   'trip_id' => $tripID
+                                               ], [
+                                                   'category'    => $tripJson->line->product,
+                                                   'number'      => $tripJson->line->id,
+                                                   'linename'    => $tripJson->line->name,
+                                                   'origin'      => $origin->ibnr,
+                                                   'destination' => $destination->ibnr,
+                                                   'stopovers'   => json_encode($tripJson->stopovers),
+                                                   'polyline'    => $polylineHash,
+                                                   'departure'   => $tripJson->plannedDeparture,
+                                                   'arrival'     => $tripJson->plannedArrival,
+                                                   'delay'       => $tripJson->arrivalDelay ?? null
+                                               ]);
+
+        foreach ($tripJson->stopovers as $stopover) {
+            $hafasStop = self::parseHafasStopObject($stopover->stop);
+
+            //This array is a workaround because Hafas doesn't give
+            //us delay-data if the train already passed this station
+            //so.. just save data we really got. :)
+            $updatePayload = [];
+
+            if ($stopover->plannedArrival != null) {
```

Comment:
You can remove every `null`-check and directly set the variables if you remove every null-value with `array_filter($arr, static function($var){return $var !== null;} );` or sth. else